File: file-service/src/main/java/net/blueshell/fileservice/repository/FileRepository.java
package net.blueshell.fileservice.repository;

import net.blueshell.db.BaseRepository;
import net.blueshell.fileservice.model.File;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface FileRepository extends BaseRepository<File, Long> {
    Optional<File> findByName(String name);
}

File: api/src/main/kotlin/net/blueshell/api/domain/file/persistence/repository/FileRepository.kt
package net.blueshell.api.domain.file.persistence.repository

import net.blueshell.api.file.persistence.File
import net.blueshell.api.shared.repository.BaseRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Suppress("FunctionName")
@Repository
interface FileRepository : BaseRepository<File, Long> {
    fun findByName(name: String): Optional<File>

    fun findFirstByEventBannersIdEventId(eventId: Long): Optional<File>

    fun findByPath(path: String): Optional<File>
}